index.js

```diff
@@ -2,91 +2,119 @@
 
 'use strict';
 
-var tls = require('tls');
-var net = require('net');
-var eos = require('end-of-stream');
-var through = require('through2');
-var minimist = require('minimist');
-var allContainers = require('docker-allcontainers');
-var statsFactory = require('docker-stats');
-var logFactory = require('docker-loghose');
-var eventsFactory = require('docker-event-log');
-var os = require('os');
+const tls = require('tls');
+const net = require('net');
+const eos = require('end-of-stream');
+const through = require('through2');
+const minimist = require('minimist');
+const allContainers = require('docker-allcontainers');
+const statsFactory = require('docker-stats');
+const logFactory = require('docker-loghose');
+const eventsFactory = require('docker-event-log');
+const os = require('os');
+
+let debugLogging = !!process.env.INSIGHT_DOCKER_DEBUG;
+
+const logDebug = (...args) => {
+  if (!debugLogging) {
```

Comment:
shorter
```suggestion
  if (debugLogging) {
    console.log(...args);
  }
```

Reply:
worse 🤷‍♂

Reply:
what?

Reply:
It's clearer to check preconditions first, and then perform the requested action. Matter of opinion, really.